checkov/terraform/graph_builder/variable_rendering/evaluate_terraform.py

```diff
@@ -167,6 +172,85 @@ def evaluate_compare(input_str: str) -> Union[str, bool]:
     return input_str
 
 
+def handle_for_loop_in_foreach(input_str: str) -> str:
+    # not implementing for loop with condition for now
+    if isinstance(input_str, str) and renderer.FOR_LOOP in input_str and '?' not in input_str:
+        old_input_str = input_str
+        try:
+            e = ast.literal_eval(input_str)
+            if isinstance(e, list) and len(e) == 1:
+                input_str = e[0]
+        except (ValueError, SyntaxError):
+            pass
+        if isinstance(input_str, str) and renderer.FOR_LOOP in input_str:
+            if input_str.startswith(f'{renderer.DOLLAR_PREFIX}{renderer.LEFT_CURLY}') and input_str.endswith(renderer.RIGHT_CURLY):
+                input_str = input_str[2:-1]
+
+            start_bracket_idx = input_str[1:].find(renderer.LEFT_BRACKET)
+            end_bracket_idx = renderer.find_match_bracket_index(input_str, start_bracket_idx + 1)
+            if start_bracket_idx == -1 or end_bracket_idx == -1:
+                return old_input_str
+
+            rendered_foreach_statement = input_str[start_bracket_idx:end_bracket_idx + 1].replace('"', '\\"').replace("'", '"')
+            new_val = ''
+            if input_str.startswith(renderer.LEFT_CURLY):
+                try:
+                    rendered_foreach_statement = json.loads(rendered_foreach_statement)
+                except JSONDecodeError:
+                    return old_input_str
+                new_val = _handle_for_loop_in_dict(rendered_foreach_statement, input_str, end_bracket_idx + 1)
+            elif input_str.startswith(renderer.LEFT_BRACKET):
+                try:
+                    rendered_foreach_statement = ast.literal_eval(rendered_foreach_statement.replace(' ', ''))
+                except (ValueError, SyntaxError):
+                    return old_input_str
+                new_val = _handle_for_loop_in_list(rendered_foreach_statement, input_str, end_bracket_idx + 1)
+            return new_val if new_val else old_input_str
+        else:
+            return input_str
+    else:
+        return input_str
+
+
+def _extract_expression_from_statement(statement: str, start_expression_idx: int) -> str:
+    """
+    statement: [ for val in ["v", "k"] : val ]
```

Comment:
I'm not sure I understand this documenation.
Also I assume from the previous CR that `3` refers to `len(' : ')`.
In my opinion it's better to create a const like `KEY_VALUE_SEPERATOR = ' : '`, and then use `len(KEY_VALUE_SEPERATOR)`